processor/probabilisticsamplerprocessor/sampler_mode.go

```diff
@@ -0,0 +1,316 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package probabilisticsamplerprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/sampling"
+)
+
+const (
+	// Hashing method: The constants below help translate user friendly percentages
+	// to numbers direct used in sampling.
+	numHashBucketsLg2     = 14
+	numHashBuckets        = 0x4000 // Using a power of 2 to avoid division.
+	bitMaskHashBuckets    = numHashBuckets - 1
+	percentageScaleFactor = numHashBuckets / 100.0
+)
+
+type SamplerMode string
```

Comment:
Do you intend on making this public for the world? If not, perhaps move it to the internal package of this module?

Reply:
This component appears to be used from other components in the repository. I suspect it should be an exported symbol--I've added a comment.

```
// SamplerMode controls the logic used in making a sampling decision.
// The HashSeed mode is the only mode, presently, and it is also the
// default mode.
//
// TODO: In the future, when OTEP 235 is introduced, there will be two
// new modes.
```

See 7abdf2d450.